src/base/net/geoipmanager.cpp

```diff
@@ -110,13 +111,20 @@ void GeoIPManager::loadDatabase()
 
 void GeoIPManager::manageDatabaseUpdate()
 {
+    if (m_dbCompleteUrl.isEmpty()) {
+        if (!m_geoIPDatabase)
+            Logger::instance()->addMessage(tr("No GeoIP license key has been configured nor a GeoIP database file has been found."), Log::WARNING);
```

Comment:
IMO, log message in wrong place (under wrong condition). When user doesn't provide license key we shouldn't even trigger database update.
The only valid error case is when user enables "resolve peer country" feature but there are no license key or db file provided.

Reply:
How about this one?

Reply:
Hmm, I am not sure I agree with you. `manageDatabaseUpdate()` "manages" the db update. Should calling code do the error checking for it? My position is that the error checking must be done by `manageDatabaseUpdate()` itself.

Reply:
My comment is more about "what we should consider as an error" than "what place in the code we should handle this error".

Reply:
Possible algorithm (when "resolve peer country" is enabled):
1. If license key is set by user try to update database and log errors if they are. Otherwise just skip database update step.
2. If database isn't loaded by previous step try to load it from file and log errors if they are.

Reply:
Now I understand your objection better. I pushed an update and hopefully manages things more fully and clearer.
PS: The logic is to first load any existing db and then perform db update if possible and necessary.